solr/core/src/java/org/apache/solr/cli/SolrCLI.java

```diff
@@ -383,6 +383,11 @@ public static boolean exceptionIsAuthRelated(Exception exc) {
   }
 
   public static SolrClient getSolrClient(String solrUrl) {
+    // today we require all urls to end in /solr, however in the future we will need to support the
+    // /api url end point instead.
+    if (!solrUrl.endsWith(("/solr"))) {
```

Comment:
[0] Not sure whether this is intended or not, but if the provided URL was `http://localhost:8983/solr/` (not the ending '/') we'd still append`/solr`.  I wonder if there isn't some Java utility for adding path segments to a URL that can be a little smarter about some of the edge cases (handling adjacent forward slashes, etc.)

Reply:
So this kind of made me sad...  basically I was trying to write both code to handle legacy /solr type urls, but then, since everything TODAY requries a /solr url, then to inject it here.   Have we figured out how in the future to hit a V2 api that has an /api end point instead via a SolrClient?    My hope is once that is solved then we remove this logic.  
Having said that, if there is better code to use here, would love some of that!   I hope that before 10x gets released, this code will be eliminated, so I kind of just put this in "for now"....    I don't expect to backport this to 9x.

Reply:
Ugh...  One test has "http://127.0.0.1:51902/uf_" as the URL.....

Reply:
Oh wait...  why do tests now have solr urls that end in `/solr`???    Now I wonder if they also stand up Solr servers that don't use `/api` for the v2 api endpoint?

Reply:
okay, I've pulled in `main` that has the now hardcoded hostContext...